executor/index_lookup_hash_join.go

```diff
@@ -294,7 +394,14 @@ func (iw *indexHashJoinInnerWorker) run(ctx context.Context, cancelFunc context.
 			joinResult.err = task.err
 			break
 		}
-		err := iw.handleTask(ctx, cancelFunc, task, joinResult, h)
+		if task.keepOuterOrder {
+			if resultCh != nil {
+				close(resultCh)
+				resultCh = nil
```

Comment:
This line seems to be useless.

Reply:
It's used to avoid `resultCh` to be closed again.